tests/unit/test_data_plane.py

```diff
@@ -418,26 +428,27 @@ def test_query_simple_with_values_mixed_metadata(test_data_plane_index):
         else:
             assert not match_vector.metadata
 
-# not deterministic
-# def test_query_simple_with_filter_values_metadata(test_data_plane_index):
-#     index, _ = test_data_plane_index
-#     namespace = 'test_query_simple_with_filter_values_metadata'
-#     vector_count = 10
-#     test_data = write_test_data(index, namespace, vector_count)
-#     # simple query - with filter, with data, with metadata
-#     api_response = index.query(
-#         values=[0.1] * vector_dim,
-#         namespace=namespace,
-#         top_k=10,
-#         include_values=True,
-#         include_metadata=True,
-#         filter={'genre': {'$in': ['action']}}
-#     )
-#     logger.debug('got query (with filter, with data, with metadata) response: {}', api_response)
-#     first_match_vector = api_response[0]
-#     assert first_match_vector.values == test_data.get(first_match_vector.id).values
-#     assert first_match_vector.metadata == test_data.get(first_match_vector.id).metadata
-#     assert first_match_vector.metadata.get('genre') == 'action'
+
+def test_query_simple_with_filter_values_metadata(test_data_plane_index):
+    index, _ = test_data_plane_index
+    namespace = 'test_query_simple_with_filter_values_metadata'
+    vector_count = 10
+    test_data = write_test_data(index, namespace, vector_count)
+    api_response = index.query(
+        values=[0.1] * vector_dim,
+        namespace=namespace,
+        top_k=10,
+        include_values=True,
+        include_metadata=True,
+        filter={'genre': {'$in': ['action']}}
+    )
+    logger.debug('got query (with filter, with data, with metadata) response: {}', api_response)
+    if not api_response:
+        pytest.skip('no vectors match the filter')
```

Comment:
I think we should retry or fail if the `api_response` is not what we expect.

Reply:
This should probably be refactored so it uses its own pre-determined data, then this won't be necessary.

@rajat08 can you please take a look? I'm less familiar with this test case